### ISSUE TITLE
`x.py install` installs std N^2 times

### ISSUE THREAD

**Issue by tmandry** [rust-lang/rust#119533]
See the log below for an example, where the string `Installing stage2 std (aarch64-unknown-fuchsia)` appears 10 times. 10 is also the number of targets in my config.toml.

<details>
<summary>Log</summary>

```
Building bootstrap
    Finished dev [unoptimized] target(s) in 0.08s
WARNING: The use of `changelog-seen` is deprecated. Please refer to `change-id` option in `config.example.toml` instead.
WARNING: The `change-id` is missing in the `config.toml`. This means that you will not be able to track the major changes made to the bootstrap configurations.
NOTE: to silence this warning, add `change-id = 119373` at the top of `config.toml`
Building stage0 library artifacts (x86_64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.17s
Building compiler artifacts (stage0 -> stage1, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.26s
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
Building stage1 library artifacts (x86_64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.16s
Building compiler artifacts (stage1 -> stage2, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.25s
Creating a sysroot for stage2 compiler (use `rustup toolchain link 'name' build/host/stage2`)
Building stage2 library artifacts (x86_64-unknown-linux-gnu -> aarch64-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.15s
Building stage0 tool rust-installer (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.14s
Dist rust-std-nightly-aarch64-unknown-linux-gnu
        finished in 2.134 seconds
Installing stage2 std (aarch64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Uplifting library (stage1 -> stage2)
Dist rust-std-nightly-x86_64-unknown-linux-gnu
        finished in 2.113 seconds
Installing stage2 std (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'
^[[1;2B
    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> thumbv6m-none-eabi)
    Finished release [optimized + debuginfo] target(s) in 0.15s
Dist rust-std-nightly-thumbv6m-none-eabi
        finished in 0.907 seconds
Installing stage2 std (thumbv6m-none-eabi)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> thumbv7m-none-eabi)
    Finished release [optimized + debuginfo] target(s) in 0.15s
Dist rust-std-nightly-thumbv7m-none-eabi
        finished in 0.916 seconds
Installing stage2 std (thumbv7m-none-eabi)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> wasm32-unknown-unknown)
warning: dropping unsupported crate type `dylib` for target `wasm32-unknown-unknown`

warning: `std` (lib) generated 1 warning
warning: `test` (lib) generated 1 warning (1 duplicate)
    Finished release [optimized + debuginfo] target(s) in 0.17s
Dist rust-std-nightly-wasm32-unknown-unknown
        finished in 1.546 seconds
Installing stage2 std (wasm32-unknown-unknown)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> riscv32imc-unknown-none-elf)
    Finished release [optimized + debuginfo] target(s) in 0.15s
Dist rust-std-nightly-riscv32imc-unknown-none-elf
        finished in 0.841 seconds
Installing stage2 std (riscv32imc-unknown-none-elf)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> riscv64gc-unknown-linux-gnu)
    Finished release [optimized + debuginfo] target(s) in 0.17s
Dist rust-std-nightly-riscv64gc-unknown-linux-gnu
        finished in 2.239 seconds
Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> aarch64-unknown-fuchsia)
    Finished release [optimized + debuginfo] target(s) in 0.18s
Dist rust-std-nightly-aarch64-unknown-fuchsia
        finished in 1.975 seconds
Installing stage2 std (aarch64-unknown-fuchsia)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> riscv64gc-unknown-fuchsia)
    Finished release [optimized + debuginfo] target(s) in 0.17s
Dist rust-std-nightly-riscv64gc-unknown-fuchsia
        finished in 2.029 seconds
Installing stage2 std (riscv64gc-unknown-fuchsia)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Building stage2 library artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-fuchsia)
    Finished release [optimized + debuginfo] target(s) in 0.17s
Dist rust-std-nightly-x86_64-unknown-fuchsia
        finished in 2.180 seconds
Installing stage2 std (x86_64-unknown-fuchsia)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (aarch64-unknown-linux-gnu)
install: uninstalling component 'rust-std-aarch64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (x86_64-unknown-linux-gnu)
install: uninstalling component 'rust-std-x86_64-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (thumbv6m-none-eabi)
install: uninstalling component 'rust-std-thumbv6m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv6m-none-eabi'

    rust std installed.

Installing stage2 std (thumbv7m-none-eabi)
install: uninstalling component 'rust-std-thumbv7m-none-eabi'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-thumbv7m-none-eabi'

    rust std installed.

Installing stage2 std (wasm32-unknown-unknown)
install: uninstalling component 'rust-std-wasm32-unknown-unknown'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-wasm32-unknown-unknown'

    rust std installed.

Installing stage2 std (riscv32imc-unknown-none-elf)
install: uninstalling component 'rust-std-riscv32imc-unknown-none-elf'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv32imc-unknown-none-elf'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-linux-gnu)
install: uninstalling component 'rust-std-riscv64gc-unknown-linux-gnu'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-linux-gnu'

    rust std installed.

Installing stage2 std (aarch64-unknown-fuchsia)
install: uninstalling component 'rust-std-aarch64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-aarch64-unknown-fuchsia'

    rust std installed.

Installing stage2 std (riscv64gc-unknown-fuchsia)
install: uninstalling component 'rust-std-riscv64gc-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-riscv64gc-unknown-fuchsia'

    rust std installed.

Installing stage2 std (x86_64-unknown-fuchsia)
install: uninstalling component 'rust-std-x86_64-unknown-fuchsia'
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-std-x86_64-unknown-fuchsia'

    rust std installed.

Uplifting rustc (stage1 -> stage3)
Building tool cargo (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.28s
Dist cargo-nightly-x86_64-unknown-linux-gnu
        finished in 0.802 seconds
Installing stage2 cargo (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'cargo'

    cargo installed.

Building tool rust-analyzer (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.26s
Dist rust-analyzer-nightly-x86_64-unknown-linux-gnu
        finished in 0.614 seconds
Installing stage2 rust-analyzer (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-analyzer-preview'
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-APACHE
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-MIT
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/README.md

    rust analyzer installed.

Building tool rustfmt (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.19s
Building tool cargo-fmt (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.18s
Dist rustfmt-nightly-x86_64-unknown-linux-gnu
        finished in 0.445 seconds
Installing stage2 rustfmt (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rustfmt-preview'
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-APACHE
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-MIT
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/README.md

    rustfmt installed.

Building stage2 tool rust-demangler (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.15s
Dist rust-demangler-nightly-x86_64-unknown-linux-gnu
        finished in 0.276 seconds
Installing stage2 rust-demangler (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-demangler-preview'
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-APACHE
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-MIT
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/README.md

    rust demangler installed.

Building tool clippy-driver (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.19s
Building tool cargo-clippy (stage2 -> stage3, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.19s
Dist clippy-nightly-x86_64-unknown-linux-gnu
        finished in 0.379 seconds
Installing stage2 clippy (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'clippy-preview'
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-APACHE
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-MIT
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/README.md

    clippy installed.

Dist llvm-tools-nightly-x86_64-unknown-linux-gnu
        finished in 3.956 seconds
Installing stage2 llvm-tools (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'llvm-tools-preview'

    llvm tools installed.

Dist rust-src-nightly
        finished in 0.795 seconds
Installing stage2 src
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rust-src'

    rust src installed.

Building tool rustdoc (stage1 -> stage2, x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.20s
Building stage1 tool rust-analyzer-proc-macro-srv (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.16s
Dist rustc-nightly-x86_64-unknown-linux-gnu
        finished in 3.377 seconds
Installing stage2 rustc (x86_64-unknown-linux-gnu)
install: creating uninstall script at /code/rust/install/fuchsia-rust/lib/rustlib/uninstall.sh
install: installing component 'rustc'
install: backing up existing file at /code/rust/install/fuchsia-rust/bin/cargo-clippy
install: backing up existing file at /code/rust/install/fuchsia-rust/bin/cargo-fmt
install: backing up existing file at /code/rust/install/fuchsia-rust/bin/clippy-driver
install: backing up existing file at /code/rust/install/fuchsia-rust/bin/rustfmt
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-APACHE
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/LICENSE-MIT
install: backing up existing file at /code/rust/install/fuchsia-rust/share/doc/rust/README.md

    rustc installed.
```

</details>

<details>
<summary>Config.toml</summary>

```toml
changelog-seen = 2

[llvm]
optimize = true
thin-lto = false
ccache = false
ninja = true
targets = "AArch64;ARM;RISCV;WebAssembly;X86"

cflags = "--sysroot=/code/sysroot/linux"
cxxflags = "--sysroot=/code/sysroot/linux -stdlib=libc++"
ldflags = "--sysroot=/code/sysroot/linux -static-libstdc++"

use-libcxx = true
# TODO(https://github.com/rust-lang/rust/issues/94983)
# We shouldn't have to set this to false to enable statically linking libc++
static-libstdcpp = false

[build]
target = ["aarch64-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "thumbv6m-none-eabi", "thumbv7m-none-eabi", "wasm32-unknown-unknown", "riscv32imc-unknown-none-elf", "riscv64gc-unknown-linux-gnu", "aarch64-unknown-fuchsia", "riscv64gc-unknown-fuchsia", "x86_64-unknown-fuchsia"]

build-dir = "fuchsia-build"
submodules = false
docs = false
extended = true
cargo-native-static = true
tools = ["cargo", "rustfmt", "cargo-fmt", "clippy", "src", "rust-demangler", "llvm-tools", "rust-analyzer", "rustdoc"]

# Use our own compiler as stage0 for bootstrapping, rather than downloading
cargo = "/code/stage0/bin/cargo"
rustc = "/code/stage0/bin/rustc"
rustfmt = "/code/stage0/bin/rustfmt"


[install]
prefix = "/code/rust/install/fuchsia-rust"
sysconfdir = "etc"

[rust]
optimize = true
codegen-units-std = 1
channel = "nightly"
lld = false
llvm-libunwind = "in-tree"
jemalloc = true


# Enable full debuginfo for libstd.
debuginfo-level-std = 2

[dist]

[target.aarch64-unknown-linux-gnu]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = true
llvm-has-rust-patches = true

[target.x86_64-unknown-linux-gnu]
cc = "/tmp/tmpj9zh2y2k"
cxx = "/tmp/tmp0qphkowe"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/tmp/tmpe5eji4ux"
profiler = true
llvm-has-rust-patches = true

[target.thumbv6m-none-eabi]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = true
llvm-has-rust-patches = true

[target.thumbv7m-none-eabi]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = true
llvm-has-rust-patches = true

[target.wasm32-unknown-unknown]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = false
llvm-has-rust-patches = true

[target.riscv32imc-unknown-none-elf]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = true
llvm-has-rust-patches = true

[target.riscv64gc-unknown-linux-gnu]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/clang++"
profiler = true
llvm-has-rust-patches = true

[target.aarch64-unknown-fuchsia]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/ld.lld"
profiler = true
llvm-has-rust-patches = true

[target.riscv64gc-unknown-fuchsia]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/ld.lld"
profiler = true
llvm-has-rust-patches = true

[target.x86_64-unknown-fuchsia]
cc = "/code/clang/bin/clang"
cxx = "/code/clang/bin/clang++"
ar = "/code/clang/bin/llvm-ar"
ranlib = "/code/clang/bin/llvm-ranlib"
linker = "/code/clang/bin/ld.lld"
profiler = true
llvm-has-rust-patches = true
```

</details>
